company/okrs/2020_q1.md

```diff
@@ -0,0 +1,91 @@
+
+## Engineering
+
+1. **VP Eng: Build a world class engineering team.**
+   1. VP Eng KR: Grow team to plan (+1 code intel, +2 distribution, +2 web, +1 core services).
+   1. VP Eng KR: Transition an existing teammate to be (or hire) an Engineering Manager of the distribution team.
+   1. VP Eng KR: Document how we do performance reviews and determine compensation for teammates in engineering roles.
+1. **VP Eng: Create a data driven planning process.**
+   1. VP Eng KR: Each item in a tracking issue has an estimate of time or size.
+   1. VP Eng KR: Each team commits to a written work plan for each release cycle that accounts for the work estimates and the team's capacity.
+   1. VP Eng KR: Each team completes >80% of the work that they committed to at the beginning of the release cycle each iteration.
+
+1. **Distribution: Resolve issues for critical customer deals and expansions**
+1. **Distribution: Improve developer productivity**
+   1. Distribution KR: Lower build times to X
+   1. Distribution KR: Automate away release captain role
+   1. Distribution KR: Infrastructure for automated deploys and releases (no human in loop required by default)
+1. **Distribution: Build features that accelerate customers through pipeline**
+   1. Distribution KR: Shorten the time from "intent to try Sourcegraph" to "whole org is using Sourcegraph" (onboarding experience)
+   1. Distribution KR: Ensure new features (LSIF, GitLab Native Integration, Automation) are accessible and easily tryable by customers once they go GA
+   1. Distribution KR: Surface information to site admins and managers to encourage expansion/upsell
+
+1. **Code intelligence: LSIF is adopted by customers**
+   1. Code intelligence KR: At least three customers are using LSIF on their primary repositories
+   1. Code intelligence KR: At least one customer is using LSIF across all of their code in languages for which there exists an LSIF indexer
+
+1. **Code intelligence: Increase user confidence in code intelligence provided by LSIF**
+   1. Code intelligence KR: Code intelligence UI (tooltips, references panel, etc.) indicates whether or not results are precise
+   1. Code intelligence KR: Fall back to fuzzy if the file has been modified since the last LSIF upload
+   1. Code intelligence KR: Make use of LSIF data when the current line is not part of the diff since the last LSIF upload
+
+1. **Code intelligence: Make search better using precise code intel data**
+   1. Code intelligence KR: Implement search within dependencies and/or dependents of current repo
+   1. Code intelligence KR: Implement searching for defs/refs by symbol name
+
+1. **Web: Make users aware of advanced search features**
+   1. Web KR: Start tracking time to first search and search filters per query in pings
+   1. Web KR: Shorten time to first search
+   1. Web KR: Increase average number of search filters per query
+
+1. **Web: Automation is adopted by customers**
```

Comment:
@nicksnyder @christinaforney @mrnugget @tsenart @felixfbecker how would you feel about this automation OKR? (It should probably be for both Web and Core Services @tsenart)

Reply:
> At least four customers are using campaigns in production

I don't know whether four is realistic or even within reach, but I like "customer using in production" :)

The one thought I had though is that ultimately this is not something we can influence, right? It's still up to a customer to decide to use. I don't know if that makes a difference, but at previous companies where we used OKRs one guideline was that we should only define Os and KRs that we could, theoretically, complete on our own.

> All campaign types have been used by customers in production

I don't know how useful that is (also: we _could_ just reduce the number of types and have one customer use the single one that's left) and I wonder whether this doesn't put "quantity" over "quality" in a way.

Reply:
> I don't know whether four is realistic

This was an arbitrary starting point. Open to edits on this :) My thinking was that this is something we could influence, through:
- Helping customers who are keen to try it out to set up a8n, create their first campaigns. We would probably get valuable, actionable feedback this way, although I know that a8n has already been demoed/discussed with customers a lot.  
- Identifying and solving blockers / feature gaps that are preventing customers from trying it out.

Do you have suggestions for other KRs that we could more directly make progress on, without being blocked on externalities, and that wouldn't amount to "build features X, Y, Z"?

> I don't know how useful that is

On second thought I think you're right, I'll remove this one.

Reply:
> You can certainly influence it by focusing on building things that will unblock particular customers from actually using automation (and not working on things that don't directly contribute to us getting customers). I hope we can agree that the outcome we want as a company is to get customers on automation. Given that, it is just a matter of who should own that outcome.

Absolutely. "Influence" was probably the wrong word here. What I meant to say: even if we get four customers to say "yes, we'll purchase Automation and yes, we'll use it in production" in the end it's still possible for deal/rollout to be delayed due to external factors, such as customer not getting contract approved by finance department, etc.

I want the goals (1) to be as explicit as possible and (2) avoid measuring the wrong thing. If we have to add "well, what we mean with that is that we do everything _in our power_ to enable customers to adopt Automation" then I'd rather have "achieve $everything-in-our-power" as the key result. Things like: remove feature flag for Automation (meaning: get it polished!), write "getting started with Automation" guide, do 5 customer demos for Automation, record an Automation screencast, ...

> I think "don't build campaign types unless we can get them used by a customer in production" is a valid way to interpret this KR. Focus is good.

Same thing here. I don't think "All campaign types have been used by customers in production" is a key result that _we as engineers_ can check off. We can do exactly what you said and "don't build campaign types unless we can get them used by a customer in production", but I was hoping there's a better way to formulate it.

> Do you have suggestions for other KRs that we could more directly make progress on, without being blocked on externalities, and that wouldn't amount to "build features X, Y, Z"?

See my list in the paragraph above. I think we mainly need to close the feature gap (drafts, updating campaigns, stacking diffs, campaign timeline events, ...), polish (little UX polishes, _great_ UI for managing campaigns, stability, performance) and document it.